app/models/container_project.rb

```diff
@@ -110,4 +110,95 @@ def self.raise_creation_events(container_project_ids)
   def raise_creation_event
     MiqEvent.raise_evm_event(self, 'containerproject_created', {})
   end
+
+  def self.class_by_ems(ext_management_system)
+    ext_management_system&.class_by_ems(:ContainerProject)
+  end
+
+  def self.create_container_project(ems_id, options = {})
+    ems = ExtManagementSystem.find_by(:id => ems_id)
+    raise ArgumentError, _("EMS cannot be nil") if ems.nil?
+
+    klass = ems.class_by_ems(:ContainerProject)
+    klass.raw_create_container_project(ems, options)
+  end
+
+  def self.raw_create_container_project(_ext_management_system, _options = {})
+    raise NotImplementedError, _("raw_create_container_project must be implemented in a subclass")
+  end
+
+  def self.create_container_project_queue(userid, ext_management_system, options = {})
+    task_opts = {
+      :action => "Creating Container Project for user #{userid}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => ext_management_system.class_by_ems(:ContainerProject).name,
+      :method_name => 'create_container_project',
+      :priority    => MiqQueue::HIGH_PRIORITY,
```

Comment:
I know this was mostly based on OpenStack operations, but I don't know why they use HIGH_PRIORITY either and I think we can drop this
```suggestion
```